### PR TITLE
update hostgroup webui tests with negative scenarios

### PR DESCRIPTION
**Pull request by pavelpicka**
Automatization of hostgroups negative scenarios and all buttons test.

### CODE REVIEW

**Comment by felipevolpone** [comment]
Hello @pavelpicka, you have a duplicated section on `freeipa-pr-ci` file (the `ipaserver` one). Once you remove it, the tests will start to run.

File: ipatests/test_webui/test_hostgroup.py [hunk 1]
```diff
@@ -31,6 +31,15 @@
 import pytest
 
 
+def test_invalid_names(self, names, error_link):
+    for name in names:
+        self.navigate_to_entity(hostgroup.ENTITY)
```
Comment (pvoborni) [inline ipatests/test_webui/test_hostgroup.py]: the `navigate` line could be above the for loop, as only one navigation should happen, right?

File: ipatests/test_webui/test_hostgroup.py [hunk 2]
```diff
@@ -142,7 +151,110 @@ def test_indirect_associations(self):
...
+
+        # test to rewrite invalid input_type
+        self.navigate_to_entity(hostgroup.ENTITY)
+        self.button_click('add')
+        self.fill_input('cn', hostgroup.LEADING_SPACE)
+        self.fill_input('cn', hostgroup.PKEY3)
+        self.dialog_button_click('add')
+        self.wait_for_request(n=3)
+
+        self.navigate_to_entity(hostgroup.ENTITY)
```
Comment (pvoborni) [inline ipatests/test_webui/test_hostgroup.py]: is this navigation required here?

File: ipatests/test_webui/test_hostgroup.py [hunk 3]
```diff
@@ -142,7 +151,110 @@ def test_indirect_associations(self):
...
+        self.dialog_button_click('cancel')
+
+        self.select_record(hostgroup.PKEY)
+        self.button_click('remove')
+        self.dialog_button_click('cancel')
+        self.wait()
+        self.select_record(hostgroup.PKEY, unselect=True)
+
+        # test to rewrite invalid input_type
+        self.navigate_to_entity(hostgroup.ENTITY)
```
Comment (pvoborni) [inline ipatests/test_webui/test_hostgroup.py]: is this navigation required here?

**Comment by pavelpicka** [comment]
fixed as commented, good points

File: ipatests/test_webui/test_hostgroup.py [hunk 4]
```diff
@@ -31,6 +31,15 @@
 import pytest
 
 
+def test_invalid_names(self, names, error_link):
```
Comment (pvoborni) [inline ipatests/test_webui/test_hostgroup.py]: Test results show that this fail. The method has "test" in a name, which means that pytest picks it as something to run. IMO, the method needs to be renamed in order the test suite to succeed.

**Comment by pavelpicka** [comment]
helper function name changed from *test* to not be taken as a test

**Comment by pavelpicka** [comment]
temp commit for PR-CI removed

**Comment by tiran** [comment]
The test failure is unrelated to your change. It's about time we kill NTPD. It's starting to segfault on us:

```
Mar 26 07:54:17 master.ipa.test systemd[1]: Started Process Core Dump (PID 15566/UID 0).
Mar 26 07:54:17 master.ipa.test systemd[1]: ntpd.service: Control process exited, code=killed status=11
Mar 26 07:54:17 master.ipa.test systemd[1]: Failed to start Network Time Service.
Mar 26 07:54:17 master.ipa.test systemd[1]: ntpd.service: Unit entered failed state.
Mar 26 07:54:17 master.ipa.test audit[1]: SERVICE_START pid=1 uid=0 auid=4294967295 ses=4294967295 subj=system_u:system_r:init_t:s0 msg='unit=ntpd comm="systemd" exe="/usr/lib/systemd/systemd" hostname=? addr=? terminal=? res=failed'
Mar 26 07:54:17 master.ipa.test systemd[1]: ntpd.service: Failed with result 'signal'.
Mar 26 07:54:17 master.ipa.test systemd-coredump[15567]: Process 15564 (ntpd) of user 0 dumped core.
                                                         
                                                         Stack trace of thread 15565:
                                                         #0  0x00007f423670b0c1 _dl_name_match_p (ld-linux-x86-64.so.2)
                                                         #1  0x00007f423670349c do_lookup_x (ld-linux-x86-64.so.2)
                                                         #2  0x00007f423670433f _dl_lookup_symbol_x (ld-linux-x86-64.so.2)
                                                         #3  0x00007f42367095b3 _dl_fixup (ld-linux-x86-64.so.2)
                                                         #4  0x00007f423671148a _dl_runtime_resolve_xsavec (ld-linux-x86-64.so.2)
                                                         #5  0x00007f4234c68531 _Unwind_Find_FDE (libgcc_s.so.1)
                                                         #6  0x00007f4234c64a73 n/a (libgcc_s.so.1)
                                                         #7  0x00007f4234c65ce0 n/a (libgcc_s.so.1)
                                                         #8  0x00007f4234c66486 _Unwind_ForcedUnwind (libgcc_s.so.1)
                                                         #9  0x00007f42352611c0 __pthread_unwind (libpthread.so.0)
                                                         #10 0x00007f4235255ca2 sigcancel_handler (libpthread.so.0)
                                                         #11 0x00007f4235262a80 __restore_rt (libpthread.so.0)
                                                         #12 0x00007f4234f48ad0 __nanosleep (libc.so.6)
                                                         #13 0x00007f4234f489da sleep (libc.so.6)
                                                         #14 0x000055c6765dedf2 my_pthread_warmup_worker (ntpd)
                                                         #15 0x00007f423525761b start_thread (libpthread.so.0)
                                                         #16 0x00007f4234f8491f __clone (libc.so.6)
                                                         
                                                         Stack trace of thread 15564:
                                                         #0  0x00007f4235258b7d pthread_join (libpthread.so.0)
                                                         #1  0x000055c6765defd4 ntpdmain (ntpd)
                                                         #2  0x00007f4234e8e00a __libc_start_main (libc.so.6)
                                                         #3  0x000055c6765cf17a _start (ntpd)
```

**Comment by tiran** [comment]
master:

* 1fe795b75b9b3b123f055fcd620dd45c2535673f WebUI Hostgroups tests cases added

**Pull request closed without merging**